File: algorithms/search-algorithms/binary-search/java/BinarySearch.java
public class BinarySearch {
    public static void main(String[] args) {
        int[] arr = { 2, 4, 6, 8, 10, 12, 14, 16 };
        int target = 10;

        int result = binarySearch(arr, target);

        if (result == -1) {
            System.out.println("Target element not found.");
        } else {
            System.out.println("Target element found at " + result + ".");
        }
    }

    public static int binarySearch(int[] array, int target) {
        int low = 0;
        int high = array.length - 1;

        while (low <= high) {
            int mid = (low + high) / 2;

            if (array[mid] == target) {
                return mid;
            } else if (array[mid] < target) {
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }

        return -1;
    }
}

File: algorithms/search-algorithms/binary-search/python/binarySearch.py
def binary_search(array, target):
    low = 0
    high = len(array) - 1

    while low <= high:
        mid = (low + high) // 2

        if array[mid] == target:
            return mid
        elif array[mid] < target:
            low = mid + 1
        else:
            high = mid - 1

    return -1


arr = [2, 4, 6, 8, 10, 12, 14, 16]
target = 10

result = binary_search(arr, target)

if result == -1:
    print("Target not found")
else:
    print(f"Target found at index {result}")
